tgui/packages/tgui-panel/settings/settingsImExport.ts

```diff
@@ -0,0 +1,59 @@
+import { useDispatch } from 'tgui/backend';
+
+import { Page } from '../chat/types';
+import { importSettings } from './actions';
+
+export function exportChatSettings(
+  settings: Record<string, any>,
+  pages: Record<string, Page>[],
+) {
+  const opts: SaveFilePickerOptions = {
+    id: `ss13-chatprefs-${Date.now()}`,
+    suggestedName: `ss13-chatsettings-${new Date().toJSON().slice(0, 10)}.json`,
+    types: [
+      {
+        description: 'SS13 file',
+        accept: { 'application/json': ['.json'] },
+      },
+    ],
+  };
+
+  const pagesEntry: Record<string, Page>[] = [];
+  pagesEntry['chatPages'] = pages;
+
+  const exportObject = Object.assign(settings, pagesEntry);
+
+  window
+    .showSaveFilePicker(opts)
+    .then((fileHandle) => {
+      try {
+        fileHandle.createWritable().then((writableHandle) => {
+          writableHandle.write(JSON.stringify(exportObject));
+          writableHandle.close();
+        });
+      } catch (e) {
+        console.error(e);
+      }
+    })
+    .catch((e) => {
```

Comment:
we're mixing `.then` with `try`, can this be simplified to use one or the other?